packages/i18n/src/create-i18n.js

```diff
@@ -158,9 +158,19 @@ export const createI18n = ( initialData, initialDomain, hooks ) => {
 
 	/**
 	 * @param {LocaleData} [data]
-	 * @param {string} [domain]
+	 * @param {string}     [domain]
+	 * @param {boolean}    [shouldMerge]
 	 */
-	const doSetLocaleData = ( data, domain = 'default' ) => {
+	const doSetLocaleData = (
+		data,
+		domain = 'default',
+		shouldMerge = true
+	) => {
+		if ( ! shouldMerge ) {
+			// Remove the existing locale data for the specified domain.
+			delete tannin.data[ domain ];
+		}
```

Comment:
This is one place where `setLocaleData( data, domain, false )` is not equivalent to recreating the `I18n` instance. Data for other domains would remain in place.

Given that the use case that led to this new API is changing the language of the instance, it would make more sense to remove other domains' data, too.